### PR TITLE
Fix SQL connection pool exhaustion issues

### PR DESCRIPTION
**Pull request by azygis**
Many times we've been hit by the pool exhaustion exceptions when running the hangfire instance using this provider. Changes in this PR are somewhat copied over from Hangfire.SqlServer provider. Idea is to use one connection for a single resource.

We've been running this DLL for 4 months now and haven't experienced another exception related to connection pool since then.

### CODE REVIEW

File: src/Hangfire.PostgreSql/ExpirationManager.cs [hunk 1]
```diff
@@ -78,36 +78,30 @@ public void Execute(CancellationToken cancellationToken)
...
+                        removedCount = connection.Execute(
+                            string.Format(@"
 DELETE FROM """ + _storage.Options.SchemaName + @""".""{0}"" 
 WHERE ""id"" IN (
     SELECT ""id"" 
     FROM """ + _storage.Options.SchemaName + @""".""{0}"" 
     WHERE ""expireat"" < NOW() AT TIME ZONE 'UTC' 
     LIMIT {1}
-)", table, _storage.Options.DeleteExpiredBatchSize.ToString(CultureInfo.InvariantCulture)), transaction);
+)", table, _storage.Options.DeleteExpiredBatchSize.ToString(CultureInfo.InvariantCulture)));
```
Comment (frankhommers) [inline src/Hangfire.PostgreSql/ExpirationManager.cs:95]: Why did you put this outside the transaction?

**Review by frankhommers — CHANGES_REQUESTED** [review]

File: src/Hangfire.PostgreSql/ExpirationManager.cs (hunk 1 above)
Comment (azygis) [reply src/Hangfire.PostgreSql/ExpirationManager.cs:95]: A honest whoops of trial and error, I think. Will bring it back tomorrow.

Comment (azygis) [reply src/Hangfire.PostgreSql/ExpirationManager.cs]: Apparently transaction was never actually used. It was passed as query parameter and not as transaction:
image
Anyway, fixed that as well.

**Comment by frankhommers** [comment]
Just another question in PostgreSqlConnection:

GetRangeFromSet and GetRangeFromList

Why couldn't those be done with a LIMIT and OFFSET ?

**Comment by azygis** [comment]
Sorry, I couldn't say even if I wanted to. I have not changed any of the queries, only how the connection is passed around and lock is used.

**Pull request merged**